cdap-ui/app/cdap/services/datasource/index.js

```diff
@@ -19,20 +19,31 @@ import uuid from 'node-uuid';
 import Rx from 'rx';
 
 export default class Datasource {
-  constructor() {
+  constructor(genericResponseHandlers = [() => true]) {
     let socketData = Socket.getObservable();
     this.bindings = {};
 
     this.socketSubscription =  socketData.subscribe(
       (data) => {
         let hash = data.resource.id;
-
+        let shouldCallNextHandler = true;
         if (!this.bindings[hash]) { return; }
 
+        shouldCallNextHandler = genericResponseHandlers.reduce((prev, curr) => prev && curr(data), true);
+
         if (data.statusCode > 299 || data.warning) {
-          this.bindings[hash].rx.onError(data.error || data.response);
+          this.bindings[hash].rx.onError({
```

Comment:
so on error the handler will be called, but it will still call onError? Wouldn't this cause issue?